script/release.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+
+from collections import namedtuple
+
+import argparse
+import base64
+import copy
+import json
+import subprocess
+import sys
+import urllib.parse
+import urllib.request
+import urllib.error
+
+class Error(Exception):
+    pass
+
+class Version(object):
+    def __init__(self, version):
+        versions = version.split(sep='.')
+        if len(versions) < 2 or len(versions) > 3:
+            raise Error("Invalid version string '{}'".format(version))
+        self.major = int(versions[0])
+        self.minor = int(versions[1])
+        self.revision = int(versions[2]) if len(versions) == 3 else 0
+
+    def __str__(self):
+        return self.format()
+
+    def decrement(self):
+        if self.revision != 0:
+            self.revision -= 1
+        elif self.minor != 0:
+            self.minor -= 1
+        elif self.major != 0:
+            self.major -= 1
+        else:
+            raise Error("Cannot determine previous release")
+
+    def format(self, full = True):
+        if full or self.revision > 0:
+            return '{}.{}.{}'.format(self.major, self.minor, self.revision)
+        return '{}.{}'.format(self.major, self.minor)
+
+def verify_version(version):
+    expected = {
+        'VERSION':      [ '"{}"'.format(version), None ],
+        'VER_MAJOR':    [ str(version.major), None ],
+        'VER_MINOR':    [ str(version.minor), None ],
+        'VER_REVISION': [ str(version.revision), None ],
+        'VER_PATCH':    [ '0', None ],
+        'SOVERSION':    [ str(version.minor), None ],
```

Comment:
What should our strategy for soversion here be in a post-1.0 world?  We have a couple of choices - we could just keep it monotonically increasing and it doesn't have to have anything to do with the actual version numbers.  But I don't actually love that.

We could track it against the major/minor version numbers, which is consistent with what we've been doing - eg 0.29 -> 29, 1.0 -> 100, 1.1 -> 101.  This is the strategy that I'd advocate for.